packages/react-tinacms-editor/src/core/context/useProsemirror.tsx

```diff
@@ -0,0 +1,28 @@
+/**
+
+Copyright 2019 Forestry.io Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+import * as React from 'react'
+import { EditorView } from 'prosemirror-view'
+
+export default (): {
+  editorView: { view: EditorView } | undefined
+  setEditorView: (view: EditorView) => void
+} => {
+  const [view, setEditorView] = React.useState<EditorView>()
+  return { editorView: view ? { view } : undefined, setEditorView }
+}
```

Comment:
I like that you're disconnecting building the editor from keeping it in React.

I would suggest `useMemo` over `useState` though. It's essentially a wrapper for `useState` where you just create the object at startup time. This would makes it so `buildEditor` can just return the `view` instead of having to pass it `setEditorView`:

```suggestion
function useProsemirror(
  input: Input,
  el: HTMLDivElement | undefined | null,
  plugins: Plugin[] = [],
  imageProps: ImageProps = {},
  format?: Format
): {
  editorView: { view: EditorView } | undefined
  setEditorView: (view: EditorView) => void
} => {
  return React.useMemo(() => {
    return buildEditor(input, el, plugins, imageProps, format)
  })
}
```

Reply:
Then again, it was `useState` before. I can't remember if there was a reason for that or not. So just take this comment as a suggestion for something to investigate.

Reply:
Hey @ncphillips : `setEditorView` is needed [here](https://github.com/tinacms/tinacms/pull/1002/files#diff-9f71814e207ca3a638f57f2a4c8615e0R64).

I plan to do some more cleanup on this. I plan to move editor state creation part to context provider. So context consumers will always get latest version of `editorView`.

Reply:
Right, that’s awkward. I hope you find a nice alt